core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/DefaultAwsS3V4HttpSigner.java

```diff
@@ -15,27 +15,203 @@
 
 package software.amazon.awssdk.http.auth.aws.internal;
 
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import static software.amazon.awssdk.http.Header.CONTENT_LENGTH;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.STREAMING_UNSIGNED_PAYLOAD_TRAILER;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.UNSIGNED_PAYLOAD;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.X_AMZ_CONTENT_SHA256;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
 import software.amazon.awssdk.http.auth.aws.AwsS3V4HttpSigner;
+import software.amazon.awssdk.http.auth.aws.internal.chunkedencoding.AwsS3V4ChunkSigner;
+import software.amazon.awssdk.http.auth.aws.internal.chunkedencoding.AwsSignedChunkedEncodingInputStream;
+import software.amazon.awssdk.http.auth.aws.internal.io.AwsChunkedEncodingConfig;
+import software.amazon.awssdk.http.auth.internal.AwsV4HttpProperties;
+import software.amazon.awssdk.http.auth.internal.BaseAwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.internal.SigV4RequestContext;
+import software.amazon.awssdk.http.auth.internal.checksums.ChecksumAlgorithm;
+import software.amazon.awssdk.http.auth.internal.checksums.ContentChecksum;
+import software.amazon.awssdk.http.auth.internal.checksums.SdkChecksum;
+import software.amazon.awssdk.http.auth.internal.util.CanonicalRequestV2;
```

Comment:
More internal classes being used as protected ones. What's the expectation for this PR and next steps? Am I meant to be ignoring these problems? What *should* I be looking for?

Reply:
Everything AWS related is going to move to http-auth-aws, so there won't be cross-module.